Jenkinsfile

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env groovy
+
+pipeline {
+    agent {
+        label 'slave-group-k8s'
+    }
+
+    environment {
+        GO111MODULE = 'on'
+        KIND_VERSION = 'v0.11.0'
+        METALLB_VERSION = 'v0.9.6'
+        KUBECONFIG = "$WORKSPACE/kind-kube-config.yaml"
+        TESTING_NAMESPACE = 'namespace-for-testing'
+        WATCH_NAMESPACE = 'namespace-for-testing'
+        PATH="/opt/go/bin:$PATH"
+        RUN_SA_OPERATOR = 'true'
+        MAKE_DATADIR_WRITABLE = 'true'
+    }
+
+    options {
+        timeout(time: 60, unit: 'MINUTES')
+        timestamps()
+        buildDiscarder(logRotator(numToKeepStr: '100', daysToKeepStr: '61'))
+    }
+
+    stages {
+        stage('Build') {
+            steps {
+                sh 'go mod vendor'
+                sh 'make lint'
+            }
+        }
+
+        stage('Unit') {
+            steps {
+                sh 'make unit-test'
+            }
+        }
+
+        stage('E2E') {
+            when {
+                expression {
+                    return !env.BRANCH_NAME.startsWith('PR-') || !pullRequest.labels.contains('documentation')
```

Comment:
@ryanemerson can we take documentation based PR's logic from the Travis CI config. It's a bit smarter than labels or branch naming?

Reply:
Yes it should be. I'll see if I can get it to work.

Reply:
I haven't had time to get this working before PTO. @rigazilla @dmvolod If Travis credits run out again while I'm away, feel free to merge this PR and we can add this optimisation in the future.